docs/topics/testing/advanced.txt

```diff
@@ -67,6 +67,35 @@ The following is a simple unit test using the request factory::
             response = MyView.as_view()(request)
             self.assertEqual(response.status_code, 200)
 
+.. _topics-testing-advanced-allowed-hosts:
+
+Tests and ``ALLOWED_HOSTS``
+===========================
+
+.. versionadded:: 1.11
+
+If your application includes unit tests with custom host names, those tests
```

Comment:
It seems to me that a better solution might be to create a separate test settings file with the appropriate ALLOWED_HOSTS. Did you consider that?